### ISSUE TITLE
`prefer_final_parameters` and super parameters

### ISSUE THREAD

**Issue by Hawmex** [dart-lang/linter#3407]
If you turn on `prefer_final_parameters`, you would get complaints about not using `final` before super parameters. But I think it should be in alignment with self parameters, for which the linter doesn't complain about in a similar situation.

The code that the linter currently prefers now with `prefer_final_parameters: true`:

```dart
class A {
  final String? a;

  const A({this.a});
}

class B extends A {
  final String? b;

  const B({this.b, final super.a});
}
```

The code that I think the linter should prefer with `prefer_final_parameters: true`:

```dart
class A {
  final String? a;

  const A({this.a});
}

class B extends A {
  final String? b;

  const B({this.b, super.a});
}
```

**Comment by bwilkerson** [comment]
Adding `final` will produce a different diagnostic advising that it be removed because super parameters are implicitly final.

**Comment by Hawmex** [comment]
@asashour Sorry for the inconvinience, I misread `prefer_final_parameters` as `prefer_final_local`. I edited the title and the original comment. You should be able to reproduce the problem now.

**Comment by Hawmex** [comment]
@asashour 

I still seem to have this problem.

dart sdk version: 2.17.5

**Comment by asashour** [comment]
I guess it would take time to propagate to the Stable channel, since only critical issues are being applied to the patch releases.

Hopefully it would be in Flutter `3.1.0`, with Dart `2.18.0`.

You can test it with Flutter `master` or Dart SDK `dev` version.
